File: 04) PHP Web Development Basics/10) OOP Basics - Exercises/07. Car Salesman/Car.php
<?php

namespace Van\Classes;

use Van\Classes\Engine;

include './Engine.php';

class Car
{
    /**
     * @var string
     */
    private $model;

    /**
     * @var Engine
     */
    private $engine;

    /**
     * @var int
     */
    private $weight;

    /**
     * @var string
     */
    private $color;

    /**
     * Car constructor.
     *
     * @param string $model
     * @param Engine $engine
     */
    public function __construct(string $model, Engine $engine)
    {
        $this->model = $model;
        $this->engine = $engine;
    }

    public function __toString()
    {
        $weight = isset($this->weight)
            ? $this->weight
            : 'n/a';
        $color = isset($this->color)
            ? $this->color
            : 'n/a';

        $displacement = $this->engine->hasDisplacement()
            ? $this->engine->getDisplacement()
            : 'n/a';
        $efficiency = $this->engine->hasEfficiency()
            ? $this->engine->getEfficiency()
            : 'n/a';

        $stringToReturn = $this->model . ':' . PHP_EOL;
        $stringToReturn .= '  ' . $this->engine->getModel() . ':' . PHP_EOL;
        $stringToReturn .= '    Power: ' . $this->engine->getPower() . PHP_EOL;
        $stringToReturn .= '    Displacement: ' . $displacement . PHP_EOL;
        $stringToReturn .= '    Efficiency: ' . $efficiency . PHP_EOL;
        $stringToReturn .= '  Weight: ' . $weight . PHP_EOL;
        $stringToReturn .= '  Color: ' . $color . PHP_EOL;

        return $stringToReturn;
    }

    /**
     * @return int
     */
    public function getWeight(): int
    {
        return $this->weight;
    }

    /**
     * @param int $weight
     */
    public function setWeight(int $weight)
    {
        $this->weight = $weight;
    }

    /**
     * @return string
     */
    public function getColor(): string
    {
        return $this->color;
    }

    /**
     * @param string $color
     */
    public function setColor(string $color)
    {
        $this->color = $color;
    }
}

File: 09) Java Advanced - January 2019/14. Defining Classes - Exercises/05. Car Salesman/Car/Car.java
package Car;

import Car.Parts.Engine;

public class Car {

    private String model;

    private Engine engine;

    private int weight;

    private String color;



    public Car(String model, Engine engine) {
        setModel(model);
        setEngine(engine);
    }

    public Car(String model, Engine engine, int weight) {
        this(model, engine);
        setWeight(weight);
    }

    public Car(String model, Engine engine, int weight, String color) {
        this(model, engine, weight);
        setColor(color);
    }



    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public Engine getEngine() {
        return engine;
    }

    public void setEngine(Engine engine) {
        this.engine = engine;
    }

    public int getWeight() {
        return weight;
    }

    public void setWeight(int weight) {
        this.weight = weight;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }
}
